tests/WE2E/run_WE2E_tests.py

```diff
@@ -0,0 +1,447 @@
+    if 'USE_USER_STAGED_EXTRN_FILES' not in cfg_ics:
+        logging.debug(f'USE_USER_STAGED_EXTRN_FILES not specified in task_get_extrn_ics section of config')
+        return cfg_ics
+    if not cfg_ics['USE_USER_STAGED_EXTRN_FILES']:
+        logging.debug(f'USE_USER_STAGED_EXTRN_FILES is false for task_get_extrn_ics section of config')
+        return cfg_ics
+
+    # If EXTRN_MDL_SYSBASEDIR_ICS is "set_to_non_default_location_in_testing_script", replace with test value from machine file
+    if 'EXTRN_MDL_SYSBASEDIR_ICS' in cfg_ics:
+        if cfg_ics['EXTRN_MDL_SYSBASEDIR_ICS'] == "set_to_non_default_location_in_testing_script":
+            if 'TEST_ALT_EXTRN_MDL_SYSBASEDIR_ICS' in mach['platform']:
+                if os.path.isdir(mach['platform']['TEST_ALT_EXTRN_MDL_SYSBASEDIR_ICS']):
+                    raise FileNotFoundError(f"Non-default input file location TEST_ALT_EXTRN_MDL_SYSBASEDIR_ICS from machine file does not exist or is not a directory")
+                cfg_ics['EXTRN_MDL_SYSBASEDIR_ICS'] = mach['platform']['TEST_ALT_EXTRN_MDL_SYSBASEDIR_ICS']
+            else:
+                raise KeyError(f"Non-default input file location TEST_ALT_EXTRN_MDL_SYSBASEDIR_ICS not set in machine file")
+            return cfg_ics
+
+    # Because USE_USER_STAGED_EXTRN_FILES is true, only look on disk, and ensure the staged data directory exists
+    cfg['platform']['EXTRN_MDL_DATA_STORES'] = "disk"
+    if 'TEST_EXTRN_MDL_SOURCE_BASEDIR' not in mach['platform']:
+        raise KeyError("TEST_EXTRN_MDL_SOURCE_BASEDIR, the directory for staged test data,"\
+                       "has not been specified in the machine file for this platform")
+    if not os.path.isdir(mach['platform']['TEST_EXTRN_MDL_SOURCE_BASEDIR']):
+        raise FileNotFoundError(dedent(f"""The directory for staged test data specified in this platform's machine file
+                                TEST_EXTRN_MDL_SOURCE_BASEDIR = {mach['platform']['TEST_EXTRN_MDL_SOURCE_BASEDIR']}
+                                does not exist."""))
+
+    # Different input data types have different directory structures, so set the data directory accordingly
+    if cfg_ics['EXTRN_MDL_NAME_ICS'] == 'FV3GFS':
+        if 'FV3GFS_FILE_FMT_ICS' not in cfg_ics:
+            cfg_ics['FV3GFS_FILE_FMT_ICS'] = dflt['task_get_extrn_ics']['FV3GFS_FILE_FMT_ICS']
+        cfg_ics['EXTRN_MDL_SOURCE_BASEDIR_ICS'] = f"{mach['platform']['TEST_EXTRN_MDL_SOURCE_BASEDIR']}/"\
+                                                  f"{cfg_ics['EXTRN_MDL_NAME_ICS']}/{cfg_ics['FV3GFS_FILE_FMT_ICS']}/${{yyyymmddhh}}"
+    else:
+        cfg_ics['EXTRN_MDL_SOURCE_BASEDIR_ICS'] = f"{mach['platform']['TEST_EXTRN_MDL_SOURCE_BASEDIR']}/"\
+                                                  f"{cfg_ics['EXTRN_MDL_NAME_ICS']}/${{yyyymmddhh}}"
+
+    return cfg_ics
+
+def check_task_get_extrn_lbcs(cfg: dict, mach: dict, dflt: dict) -> dict:
+    """
```

Comment:
This is a big copy with some changes from ICS to LBCS!

Reply:
Consolidated more duplicate code into `update_expt_status()`